### PR TITLE
[MDB IGNORE] Shuttle engines part 2: Engines are now machines

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

Engines are no longer structures, instead they're machines

Now they (except large/huge ones) have circuit boards, but they can't be printed by default. Instead, this allows the machines to be destroyed and rebuilt, but maybe in the future we can add these as printable machines.
Currently the only way to get more is through Cargo so maybe we can add the boards there instead? This is really just opening the door to anyone who wishes to expand upon this.

I also removed 2 engine types, 'router' and 'platform', beacuse their sprites sucked, they had no unique mechanics, and cutting down on the amount of engines might help make the others unique just by having less to work with. They also weren't engines at all, despite what they're subtypes of.

## Why It's Good For The Game

These are already technically machines since they affect Shuttles, I thought it might be a good idea for them to be turned into machines so maybe in the future, unique functionalities can be added to them.

## Changelog

:cl:
add: Engines are now machines, not structures. If destroyed, they will drop circuit boards so you can rebuild them.
removal: In-wall engines have been remvoed. The Tram now uses full-blown engines.
/:cl:

### CODE REVIEW

**Comment by private-tristan** [comment]
will they be able to be screwdriver+crowbar'd?

**Comment by Dmeto** [comment]
would like to see them added to cargo fab, incase of the classic syndy bomb event. just make sure they aren't deleted as structures are ignored by supply shuttle's delete and the last time I checked machines weren't.

**Comment by JohnFulpWillard** [comment]
> will they be able to be screwdriver+crowbar'd?

No but it is something that can be added easily now.

> Would like to see them added to cargofab, incase of the classic syndy bomb event. 

It should still drop the boards for rebuilding, otherwise I would prefer if they were something you ordered from cargo instead of printing out at a lathe.

> Just make sure they aren't deleted, as structures are ignored by supply shuttle's delete and the last time I checked machines weren't.

The cargo shuttle sells everything on it that isn't a camera or unanchored, Shuttles start anchored so it's fine.

**Comment by CursedBirb** [comment]
Ordering them through cargo might even make sense as i dont think Nanotrasen would like to see their employes potentially dangerous crafts

**Comment by ShizCalev** [comment]
Conflicto

File: tools/UpdatePaths/Scripts/shuttle_engines.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/obj/structure/shuttle/engine/@SUBTYPES : /obj/machinery/power/shuttle_engine/@SUBTYPES{@OLD}
```
Comment (san7890) [inline tools/UpdatePaths/Scripts/shuttle_engines.txt:1]: update the name of this file to have the number of this PR so it sorts correctly?

**Comment by JohnFulpWillard** [comment]
me when i mark map edits as viewed
image

File: code/game/shuttle_engines.dm [hunk 2]
```diff
@@ -109,95 +108,88 @@
...
-/obj/structure/shuttle/engine/propulsion/burst/right
+/obj/machinery/power/shuttle_engine/propulsion/burst/right
 	name = "right burst engine"
 	icon_state = "burst_r"
 
-/obj/structure/shuttle/engine/router
-	name = "engine router"
-	icon_state = "router"
-	desc = "Redirects around energized particles in engine structures."
+/obj/machinery/power/shuttle_engine/propulsion/in_wall
```
Comment (san7890) [inline code/game/shuttle_engines.dm:156]: In_wall propulsion engines feel distinct enough to me that they shouldn't use the same circuitboard as standard propulsion engines (or at least, if you were to re-assemble it, you would get the full-sized propulsion engine anyways). Right?

Comment (JohnFulpWillard) [reply code/game/shuttle_engines.dm:156]: why do in-wall engines even exist bro i hate this place

**Comment by JohnFulpWillard** [comment]
removed in wall engines
tram looks like this now
image

i decided to make hilbert hotel look a little silly too
image

**Comment by Dmeto** [comment]
@Fikou 
As this now touches /ruins/lavaland

**Comment by JohnFulpWillard** [comment]
this is demented

**Review by Kylerace — APPROVED** [review]

**Pull request merged**